scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobState.java

```diff
@@ -265,15 +267,11 @@ private TaskStatesPage getTaskStatesPage(int offset, int limit, List<TaskState>
      * @return the list of tags.
      */
     public List<String> getTags() {
-        Set<String> result = new HashSet<>();
-        String tag = null;
-        for (TaskState task : this.getTasks()) {
-            tag = task.getTag();
-            if (tag != null) {
-                result.add(task.getTag());
-            }
-        }
-        return new ArrayList<>(result);
+        return this.getTasks().stream()
+                .map(TaskState::getTag)
+                .filter(s -> s != null)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with method reference 'Objects::nonNull'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)